lib/internal/modules/esm/loader.js

```diff
@@ -238,9 +239,15 @@ class Loader {
     });
   }
 
-  async getModuleJob(specifier, parentURL) {
+  async getModuleJob(specifier, parentURL, import_assertions) {
     const url = await this.resolve(specifier, parentURL);
     const format = await this.getFormat(url);
+
+    if (import_assertions?.type === 'json' && format !== 'json') {
```

Comment:
```suggestion
    if (import_assertions.type === 'json' && format !== 'json') {
```

Isn't it guaranteed to be an object?